docs/architecture-decisions/0005-implement-an-accessible-navigation-system.md

```diff
@@ -0,0 +1,202 @@
+a library which does support this behavior.
+
+
+### Option 1: SmartMenus 2
+
+The maintainer for SmartMenus is aware of this problem in the library, as we and
+others have reported it via GitHub (see [issue 204](https://github.com/vadikom/smartmenus/issues/204) and [issue 231](https://github.com/vadikom/smartmenus/issues/231)). He has
+committed to including this as an improvement in the future release of
+SmartMenus 2, and the documentation already available [includes this among its available
+features](https://configurator.smartmenus.org/).
+
+However, at the time of writing [SmartMenus 2 is only available via one alpha
+release](https://github.com/vadikom/smartmenus/releases) - and the maintainer has [cautioned us against using it in a production
+system yet](https://github.com/vadikom/smartmenus/issues/245).
+
+* Pros: Just upgrading our existing menu library as it improves over time is a
+  good philosophy to follow.
+
+* Cons: The timeline for the release of SmartMenus 2 is not clear, and cannot
+  be guaranteed to happen. Additionally, the significant changes to the library
+  with version 2 will likely require a significant level of effort - so the
+  benefits of a simple upgrade may not accrue based on the materials published
+  so far.
+
+
+### Option 2: Accessible Menu
+
+One alternative to SmartMenus is the [Accessible Menu](https://accessible-menu.dev/) library, which takes as its
+starting point the [ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/) for navigation menus. One of
+the supported patterns is the [Disclosure Navigation Menu with Top-Level Links](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation-hybrid/),
+which includes the needed separation between navigation links and submenu
+toggles.
+
+While the provided example includes this separation only at the top level of the
+menu, the pattern can be extended to all levels - [as demonstrated in this
+CodePen](https://codepen.io/matt-bernhardt/pen/poBRKrY).
+
+#### Benefits of Accessible Menu
+
+This menu, when properly configured, appears to implement all the best practices
+for an accessible navigation menu for web application.
+
+#### Drawbacks of Accessible Menu
+
+There are several significant drawbacks to using this menu library.
+
+* The most significant drawback is that it imposes some operating costs on users
+  of assistive technology that are not ideal. Our recent accessibility testing
+  [included some details about this problem](https://github.mit.edu/Accessibility-FY2024/omeka/issues/4#issuecomment-182190) while we discussed this menu issue.
+
+* The markup requirements for this library do not immediately align with the
+  markup generated by Omeka. While this is manageable by either maintaining a
+  set of custom navigation templates (which other themes also do) or by writing
+  some javascript to manipulate the DOM prior to instantiating the menu, it does
```

Comment:
FWIW, I'd likely lean towards maintaining custom nav templates over scripts to change the DOM before nav kicks in. Both require custom code that could break if Omeka changes upstream expectations, but the custom templates feel like they would be easier to maintain (I could be wrong, but in general fixing this as close to the sources as possible feels better to me and that's what a nav template feels like)

Reply:
For Accessible Menu, I would agree with you - the amount of customization to the markup needed for the menu to fire was best handled via custom templates rather than an intermediate javascript layer.

For the current SmartMenus system, though, that modification is extremely minimal - adding `sm` and `sm-mint` to the overall `ul` element. For that level of customization, I'd rather have [a single line of javascript](https://github.com/MITLibraries/mitlibraries-theme-omeka/blob/main/asset/js/tesseract.js#L5) than to take on [two different navigation templates](https://github.com/MITLibraries/mitlibraries-theme-omeka/tree/post-107-accessible-menu/view/common/navigation) (one for the top nav menu, and another called recursively for any submenu).

For the bespoke option we're talking about adopting, I don't think we'll be forced to make this sort of choice - but I'm happy to discuss this more.

Reply:
Yeah it's complicated at best. I guess my gut feeling is we'll eventually end up forking the nav templates into our theme so the sooner we accept that the less complicated we make this. I understand fully the desire to not fork the nav though as this is exactly the type of think that leads to annoyances when upgrading `administrate` in ETD.